src/unix.rs

```diff
@@ -111,16 +113,20 @@ impl Client {
             FromEnvErrorInner::CannotParse("expected a path after `fifo:`".to_string())
         })?;
         let path = Path::new(path_str);
+
         let file = OpenOptions::new()
             .read(true)
             .write(true)
             .open(path)
             .map_err(|err| FromEnvErrorInner::CannotOpenPath(path_str.to_string(), err))?;
 
-        Ok(Some(Client::Fifo {
-            file,
-            path: path.into(),
-            is_non_blocking: AtomicBool::new(false),
+        Ok(Some(Client {
+            read: file
+                .try_clone()
+                .map_err(|err| FromEnvErrorInner::CannotClone(file.as_raw_fd(), err))?,
```

Comment:
Introducing a public new error case just because we internally happen to clone an FD to keep two fields populated even though it's not strictly necessary does not seem like a good tradeoff.

In that case having another enum is probably better.

Reply:
I could reuse `CannotOpenFd`, does that look good to you?

Reply:
Good enough

Reply:
Reverted this change.